Detectors/Calibration/include/DetectorsCalibration/MeanVertexCalibrator.h

```diff
@@ -0,0 +1,112 @@
+    mNBinsZ(nBinsZ), mRangeZ(rangeZ), mSMAslots(nSlotsSMA) {
+    mSMAdata.init(useFit, nBinsX, rangeX, nBinsY, rangeY, nBinsZ, rangeZ);
+  }
+
+  ~MeanVertexCalibrator() final = default;
+
+  bool hasEnoughData(const Slot& slot) const final { LOG(INFO) << "container entries = " << slot.getContainer()->entries << ", minEntries = " << mMinEntries; return slot.getContainer()->entries >= mMinEntries; }
+  void initOutput() final;
+  void finalizeSlot(Slot& slot) final;
+  Slot& emplaceNewSlot(bool front, TFType tstart, TFType tend) final;
+
+  uint64_t getNSlotsSMA() const { return mSMAslots; }
+  void setNSlotsSMA(uint64_t nslots) { mSMAslots = nslots; }
+
+  void doSimpleMovingAverage(std::deque<float>& dq, float& sma);
+
+  const MVObjectVector& getMeanVertexObjectVector() const { return mMeanVertexVector; }
+  const CcdbObjectInfoVector& getMeanVertexObjectInfoVector() const { return mInfoVector; }
+  CcdbObjectInfoVector& getMeanVertexObjectInfoVector() { return mInfoVector; }
+
+ private:
+  int mMinEntries = 0;
+  int mNBinsX = 0;
+  float mRangeX = 0.;
+  int mNBinsY = 0;
+  float mRangeY = 0.;
+  int mNBinsZ = 0;
+  float mRangeZ = 0.;
+  bool mUseFit = false;
+  uint64_t mSMAslots = 5; 
+  CcdbObjectInfoVector mInfoVector; // vector of CCDB Infos , each element is filled with the CCDB description
+                                    // of the accompanying LHCPhase
+  MVObjectVector mMeanVertexVector;   // vector of Mean Vertex Objects, each element is filled in "process"
+                                      // when we finalize one slot (multiple can be finalized during the same
+                                      // "process", which is why we have a vector. Each element is to be considered
+                                      // the output of the device, and will go to the CCDB. It is the simple
+                                      // moving average
+  std::deque<float> mTmpMVobjDqX;            // This is the deque of MeanVertex objecs that will be used for the
+                                             // simple moving average, X
+  std::deque<float> mTmpMVobjDqY;            // This is the deque of MeanVertex objecs that will be used for the
+                                             // simple moving average, Y
+  std::deque<float> mTmpMVobjDqZ;            // This is the deque of MeanVertex objecs that will be used for the
+                                             // simple moving average, Z
+  std::deque<float> mTmpMVobjDqSigmaX;       // This is the deque of MeanVertex objecs that will be used for the
+                                             // simple moving average, SigmaX
+  std::deque<float> mTmpMVobjDqSigmaY;       // This is the deque of MeanVertex objecs that will be used for the
+                                             // simple moving average, SigmaY
+  std::deque<float> mTmpMVobjDqSigmaZ;       // This is the deque of MeanVertex objecs that will be used for the
```

Comment:
I thought about this option, I thought this was was easier for the moving average, I will change it. I will then change the `std::deque<MeanVertexData> mTmpMVdataDq;` to be also a deque of this struct (but I will do this next week).

Reply:
Hi @shahor02 ,
I was thinking about modifying the MeanVertexData and adding there as members the start and end, in general. What do you think? 
And: What do you think about changing the MeanVertexObject from class to struct?
Chiara

Reply:
Actually I have a second thought. I think the time should stay separate, because I use it in both cases:
- when I fit the single slot and then do the average of the results of the fits;
- when I sum up the slots, and then do the fit. 
So I will indeed change that we do not allocate one deque per data member of the MeanVertexObject, but the time (start and end) will stay separate. Let's see how it looks.

Reply:
Changed, can you check?